internal/bminventory/inventory.go

```diff
@@ -3177,3 +3177,142 @@ func proxySettingsForIgnition(httpProxy, httpsProxy, noProxy string) (string, er
+	  },
+	  "mode": 420
+	},
+	{
+	  "path": "/etc/assisted-service/nginx.conf",
+	  "contents": {
+		"compression": "gzip",
+		"source": "data:;base64,H4sIAAAAAAAC/3yQsW7zMAyEdz8FBwOZfivBnyFNxi59gc6CKtO2EEUSSCawUfjdC9keCiGNBwNHfScdj5EeSPBdAQB4x4IBTsfjf2D2l2XI7LVFEtc5awRBZcEq9C6M/xh9x64P2DaW5KlBX3H603TFaTMtOXQwNwS9jihGAWVSWqULLY7rvxnktqXz0RpxMWTQbWvkL1EcJ50MMwwi6axUJv0QWc6n/dv+UpAZ0g8kzncdmkN5zih6QNMiwWfqybQI9eK5r+oF/x5DQLuE3G307gX+EVmgzkFLyBo7oP6alq2evD4XjfyqQ2jSnfPIUN/JgSprnKu5+gkAAP//yEhoiwsCAAA="
+	  },
+	  "mode": 420
+	},
+	{
+	  "path": "/etc/assisted-service/auth.json",
+	  "contents": {
+		"source": "data:,{{.PULL_SECRET}}"
+	  },
+	  "mode": 420
+	}]
+},
+  "systemd": {
+    "units": [
+	{
+	  "contents": "[Unit]\nAfter=network-online.target\n\n[Service]\nType=oneshot\nRestart=no\nExecStart=/bin/bash /etc/assisted-service/create_certs.sh\n\n[Install]\nWantedBy=multi-user.target\n",
+	  "enabled": true,
+	  "name": "assisted-service-certs.service"
+	},
+	{
+	  "contents": "[Unit]\nAfter=network-online.target\n\n[Service]\nType=forking\nRestart=no\nExecStart=podman pod create --name assisted-installer -p 8443\n\n[Install]\nWantedBy=multi-user.target\n",
+	  "enabled": true,
+	  "name": "assisted-service-pod.service"
+	},
+	{
+	  "contents": "[Unit]\nAfter=assisted-service-pod.service\n\n[Service]\nType=forking\nRestart=no\nExecStart=podman run -dt --pod assisted-installer --env-file /etc/assisted-service/environment --authfile /etc/assisted-service/auth.json --name db quay.io/ocpmetal/postgresql-12-centos7\nTimeoutStartSec=300\n\n[Install]\nWantedBy=multi-user.target\n",
+	  "enabled": true,
+	  "name": "assisted-service-db.service"
+	},
+	{
+	  "contents": "[Unit]\nAfter=assisted-service-db.service\n\n[Service]\nType=forking\nRestart=no\nExecStart=podman run -dt --pod assisted-installer --env-file /etc/assisted-service/environment -v /etc/assisted-service/nginx-certs/nginx-selfsigned.crt:/data/nginx-selfsigned.crt:z --restart always --name installer quay.io/ocpmetal/assisted-service:latest\nTimeoutStartSec=300\n\n[Install]\nWantedBy=multi-user.target\n",
+	  "enabled": true,
+	  "name": "assisted-service-installer.service"
+	},
+	{
+	  "contents": "[Unit]\nAfter=assisted-service-installer.service\n\n[Service]\nType=forking\nRestart=no\nExecStart=podman run -dt --pod assisted-installer --env-file /etc/assisted-service/environment -v /etc/assisted-service/nginx-certs:/certs:z -v /etc/assisted-service/nginx.conf:/opt/bitnami/nginx/conf/server_blocks/nginx.conf:z --name ui quay.io/ocpmetal/ocp-metal-ui:latest\nTimeoutStartSec=300\n\n[Install]\nWantedBy=multi-user.target\n",
+	  "enabled": true,
+	  "name": "assisted-service-ui.service"
+	}
+    ]}
+}`
+
+/* DownloadLiveISO Downloads the live ISO. */
+func (b *bareMetalInventory) DownloadLiveISO(ctx context.Context, params installer.DownloadLiveISOParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+
+	sshPublicKey := params.LiveIsoCreateParams.SSHPublicKey
```

Comment:
Just wanted to understand - the sshPublicKey and pullSecret provided in this API could be different from the ones the user provides for the discovery iso?

Reply:
Yes, it can be different.